mchf-eclipse/drivers/ui/lcd/ui_lcd_items.h

```diff
@@ -9,14 +9,175 @@
 #define UI_LCD_UI_LCD_ITEMS_H_
 
 
+#define COL_SPECTRUM_GRAD					0x40
+#define Grid                RGB(COL_SPECTRUM_GRAD,COL_SPECTRUM_GRAD,COL_SPECTRUM_GRAD)      // COL_SPECTRUM_GRAD = 0x40
+#define WATERFALL_HEIGHT 70
+#define SPECTRUM_MAX_MARKER 3
+#define SPECTRUM_SCOPE_GRID_VERT_COUNT  8
+#define SPECTRUM_SCOPE_GRID_HORIZ 16
+
+
+#define LEFTBOX_WIDTH 58 // used for the lower left side controls
+#define LEFTBOX_ROW_H  (28)
+#define LEFTBOX_ROW_2ND_OFF  (13)
+
+#define ENC_COL_W (37)
+#define ENC_ROW_H (30)
+#define ENC_ROW_2ND_OFF (14)
+#define Xspacing 5
+#define ui_txt_msg_buffer_size 51		//this defines the buffer size for text line in maximum possible configuration.
+										//Please adjust it to maximum value+1 of ts.Layout->TextMsg_buffer_max
+typedef enum
+{
+    RESOLUTION_320_240,
+    RESOLUTION_480_320
+} disp_resolution_t;
+
+typedef struct {
+    uint16_t x;
+    uint16_t y;
+    uint16_t w;
+    uint16_t h;
+} UiArea_t;
+
+typedef struct
+{
+	UiArea_t region;
+	void (*function_short_press)();
+	void (*function_long_press)();
+} touchaction_descr_t;
+
+typedef struct
+{
+	const touchaction_descr_t* actions;
+	int32_t size;
+} touchaction_list_descr_t;
+
+typedef struct {
+	uint16_t SizeX;					// Lcd dimension
```

Comment:
Hi, good stuff. But if you don't mind, I'll go and change X,Y pairs to use a coord data type with x and y being members.
Reduces the number of entries a quite lot. For single X or Y we can still use uint16_t, no big problem. 

But first let the dust settle, this is a simple enough refactoring.

Reply:
Hi Danilo,
yes please do it (in fact I was waiting for your refactoring ;) ), there are more like this in this structure. And new ones will follow when we start with 800x480.

Reply:
After the today's issue with bootloader I had to add SizeX and SizeY data to  mchf_display_t becuse ts is not available in bootloader code. So we have some kind of redundant data, but for this moment I do not have idea how to (if) solve this.

Reply:
I think we should use the SizeX and SizeY or something along those lines as much as and deprecate direct use of MAX_X and MAX_Y whereever possible. And the resolution belongs to the graphics driver and not in the ts structure directly, This ts beast is way to unstructured anyway.